ui/src/components/loading.tsx

```diff
@@ -38,12 +41,24 @@ export default function Loading(props: Props) {
     return <>{props.children}</>;
   }
 
-  return <Box className="loading">{showStatus(visible, props.status)}</Box>;
+  return (
+    <Box className={props.size === "small" ? "" : "loading"}>
+      {showStatus(visible, props.status, props.size)}
+    </Box>
+  );
 }
 
-function showStatus(visible: boolean, status?: Status): ReactNode {
+function showStatus(
+  visible: boolean,
+  status?: Status,
+  size?: string
+): ReactNode {
   if (status?.error && !status?.isPending) {
-    return (
+    return size === "small" ? (
+      <Tooltip title={status.error?.message} arrow={true}>
```

Comment:
Pull the error message logic below into a variable that can be used here as well to avoid the potentially empty tooltip.

Reply:
Done.